### DISCUSSION TITLE
Availability of N-gram LM for Conformer Models

### DISCUSSION THREAD

**Discussion by agemagician** [NVIDIA-NeMo/NeMo#2619] [Q&A]
Hello,

We are trying to perform beam search decoding for stt_en_conformer_ctc_medium model, and you have reported the results for both N-gram LM and Neural Rescorer, but only the link for the Neural Rescorer is available:
https://ngc.nvidia.com/catalog/models/nvidia:nemo:stt_en_conformer_ctc_medium

Could you please share the link to the N-gram LM used for the bpe conformed models?

Thanks in advance for your reply.

**Comment by VahidooX** [comment]
We did not release that model on NGC as it is very easy and fast to train one. We trained that on the concatenation of the transcripts of LS train set with the LS language modelling corpus: https://www.openslr.org/11/

You may take a look here to see how it is trained:
https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/main/asr/asr_language_modeling.html#train-n-gram-lm

**Reply by agemagician** [reply]
Thanks for your answer.

Unfortunately, I have followed your suggestion but I can't reproduce the same results.

With the "stt_en_conformer_ctc_medium" model using CTC output, the WER for LS is 2.96%, and with the N-Gram model, it goes down to 2.78%.
This is using the same data/parameters on the "stt_en_conformer_ctc_medium" page.

On the NGC website:
https://ngc.nvidia.com/catalog/models/nvidia:nemo:stt_en_conformer_ctc_medium
The WER using the N-Gram model should be 2.5%.

It will be great if you could share the n-gram model here in the discussion.
It will help the community to reproduce the same result, as well as finding any bugs.

Thanks in advance for your reply.

**Reply by VahidooX** [reply]
Would you please share with me the exact commands you used to train and evaluate the model?
